mythril/laser/smt/bool.py

```diff
@@ -84,6 +84,7 @@ def __bool__(self) -> bool:
 def And(*args: Bool) -> Bool:
     """Create an And expression."""
     union = []
+    args = [arg if isinstance(arg, Bool) else Bool(arg) for arg in args]
```

Comment:
This is against the corresponding type hint of `Bool.And()`, will change it in #894 .